### PR TITLE
feat(eventing): nexus child events

### PR DESCRIPTION
**Pull request by datacore-vvarakantham**
Nexus child add and remove events.

### CODE REVIEW

File: io-engine/src/grpc/v1/nexus.rs [hunk 1]
```diff
@@ -572,6 +574,7 @@ impl NexusRpc for NexusService {
                 .map_err(|_| Status::cancelled("cancelled"))?
                 .map_err(Status::from)
                 .map(|nexus| {
+                    event.generate();
```
Comment (tiagolobocastro) [inline io-engine/src/grpc/v1/nexus.rs]: This should be added inline with the code itself? Otherwise if there's a timeout or disconnect we might miss this?

Comment (datacore-vvarakantham) [reply io-engine/src/grpc/v1/nexus.rs]: Yes. Will do that.

Comment (datacore-vvarakantham) [reply io-engine/src/grpc/v1/nexus.rs]: Done.

File: io-engine/src/eventing/nexus_child_events.rs
Comment (tiagolobocastro) [inline io-engine/src/eventing/nexus_child_events.rs:1]: Missing OnlineChild?

**Review by tiagolobocastro — APPROVED** [review]

**Review by hrudaya21 — APPROVED** [review]
lgtm

File: io-engine/src/bdev/nexus/nexus_bdev_children.rs [hunk 2]
```diff
@@ -66,6 +66,11 @@ use spdk_rs::{
     IoDeviceChannelTraverse,
 };
 
+
+use events_api::event::EventAction;
+
+use crate::eventing::{EventMetaGen, EventWithMeta};
```
Comment (tiagolobocastro) [inline io-engine/src/bdev/nexus/nexus_bdev_children.rs]:
```suggestion
use events_api::event::EventAction;
```

And move the "user crate" up together with the other "use crate"

**Review by tiagolobocastro — APPROVED** [review]

File: io-engine/src/bdev/nexus/nexus_bdev_children.rs (hunk 2 above)
Comment (datacore-vvarakantham) [reply io-engine/src/bdev/nexus/nexus_bdev_children.rs]: Done

**Comment by datacore-vvarakantham** [comment]
bors merge

**Pull request merged**